### PR TITLE
[tiles] add units for tiles

### PR DESCRIPTION
**Pull request by chejennifer**
- get units from data fetch for the tiles
Screenshot 2023-02-24 at 3 49 53 PM
Screenshot 2023-02-24 at 3 50 11 PM
Screenshot 2023-02-24 at 3 50 24 PM
Screenshot 2023-02-24 at 3 50 45 PM

### CODE REVIEW

File: static/js/components/tiles/bar_tile.tsx [hunk 1]
```diff
@@ -183,6 +185,11 @@ function processData(
           };
           if (raw.facets[stat.facet]) {
             sources.add(raw.facets[stat.facet].provenanceUrl);
+            let svUnit = raw.facets[stat.facet].unit;
+            if (svUnit && spec.denom) {
```
Comment (beets) [inline static/js/components/tiles/bar_tile.tsx:189]: for these, might want to check that "_Person" is in the denominator (we now also have area denominators)

File: static/js/components/tiles/line_tile.tsx [hunk 2]
```diff
@@ -188,14 +190,23 @@ function rawToChart(
           dataPoints
         )
       );
+      let svUnit = raw.facets[series.facet].unit;
+      if (svUnit && spec.denom) {
```
Comment (beets) [inline static/js/components/tiles/line_tile.tsx:194]: ditto

File: static/js/components/tiles/map_tile.tsx [hunk 3]
```diff
@@ -263,6 +266,10 @@ function processData(
   if (_.isEmpty(dataValues)) {
     return;
   }
+  let unit = getUnit(Object.values(rawData.placeStat), rawData.metadataMap);
+  if (unit && isPerCapita) {
```
Comment (beets) [inline static/js/components/tiles/map_tile.tsx:270]: ditto

**Review by beets — APPROVED** [review]
ok for today, but we'd need to fast follow and add a check for a "_Person" denominator before adding per person. might be worth pulling out into a util.

and we should add MTCO2e to [this util](https://github.com/datacommonsorg/website/blob/master/static/js/utils/string_utils.ts#L131)

**Comment by beets** [comment]
also MetricTon --> t

**Review by pradh — APPROVED** [review]
If there are short-form overrides to apply, based on the unit DCID, like `MetricTonCO2e` -> `MTCO2e`, rather than the SV... should we have some way to do it?

**Comment by pradh** [comment]
Very useful change!

**Comment by pradh** [comment]
The per-capita charts still show `%`?

**Comment by beets** [comment]
> If there are short-form overrides to apply, based on the unit DCID, like `MetricTonCO2e` -> `MTCO2e`, rather than the SV... should we have some way to do it?

yep - we do some unit rewrites in the frontend, though it's split for place explorer (internationalized) and the rest of the site...

**Comment by chejennifer** [comment]
> The per-capita charts still show `%`?

The per capita charts had their units set when building the page config .. do we want to change that?

**Comment by pradh** [comment]
> > The per-capita charts still show `%`?
> 
> The per capita charts had their units set when building the page config .. do we want to change that?

I see. We can fix as a follow on.

File: static/js/components/tiles/bar_tile.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/components/tiles/bar_tile.tsx]: done, created a util function

**Pull request merged**